src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/ProviderBase/FieldNameLookup.cs

```diff
@@ -1,159 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-namespace Microsoft.Data.ProviderBase
-{
-    using System.Collections;
-    using System.Data;
-    using System.Diagnostics;
-    using System.Globalization;
-    using Microsoft.Data.Common;
-
-    internal sealed class FieldNameLookup
-    { // V1.2.3300, MDAC 69015, 71470
-
-        // hashtable stores the index into the _fieldNames, match via case-sensitive
-        private Hashtable _fieldNameLookup;
-
-        // original names for linear searches when exact matches fail
-        private string[] _fieldNames;
-
-        // if _defaultLocaleID is -1 then _compareInfo is initialized with InvariantCulture CompareInfo
-        // otherwise it is specified by the server? for the correct compare info
-        private CompareInfo _compareInfo;
-        private int _defaultLocaleID;
-
-        public FieldNameLookup(string[] fieldNames, int defaultLocaleID)
-        { // V1.2.3300
-            if (null == fieldNames)
-            {
-                throw ADP.ArgumentNull("fieldNames");
-            }
-            _fieldNames = fieldNames;
-            _defaultLocaleID = defaultLocaleID;
-        }
-
-        public FieldNameLookup(System.Collections.ObjectModel.ReadOnlyCollection<string> columnNames, int defaultLocaleID)
-        {
-
-            int length = columnNames.Count;
-            string[] fieldNames = new string[length];
-            for (int i = 0; i < length; ++i)
-            {
-                fieldNames[i] = columnNames[i];
-                Debug.Assert(null != fieldNames[i], "MDAC 66681");
```

Comment:
I saw these `Debug.Assert` lines were removed from the constructors. Any idea what they were for?

Reply:
They make sure that there aren't column names missing. They're there because of the way that the hidden column mapping used to work and since i changed that there isn't a way to get null column named in the map, it uses an integer crossmap which is lazy initialized.